pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -438,9 +417,13 @@ func (r *ReconcileClusterDisruption) getOSDFailureDomains(clusterInfo *cephclien
 			if !osdDownFailureDomains.Has(failureDomainName) {
 				osdDownFailureDomains.Insert(failureDomainName)
 			}
-			isDrained, err := hasOSDNodeDrained(clusterInfo.Context, r.client, request.Namespace, labels[osd.OsdIdLabelKey])
+			if !downOSDs.Has(labels[osd.OsdIdLabelKey]) {
+				downOSDs.Insert(labels[osd.OsdIdLabelKey])
+			}
+			// check if OSD is down on unscheduleable node
+			isDrained, err := hasOSDNodeDrained(clusterInfo.Context, r.client, labels["topology-location-host"])
```

Comment:
Just to be clear, in the case where multiple OSDs are down, when determining which failure domain to allow draining, we only look for a node where the OSDs is `unschedulable`, correct? Nice and simple if we don't even need to check which down OSDs have 0 PGs.

Reply:
That's right. 
- If OSDs are down and the node is`unschedulable `, we treat that as a node drain event.  The failureDomain is stored in the configmap for consistency across the multiple reconciles. 
- If OSDs are down and the node is schedulable we don't treat that as a node drain event.